src/codechicken/nei/recipe/BrewingRecipeHandler.java

```diff
@@ -83,30 +97,50 @@ public String getRecipeName() {
     public void loadCraftingRecipes(String outputId, Object... results) {
-        if (outputId.equals("brewing") && getClass() == BrewingRecipeHandler.class)//don't want subclasses getting a hold of this
-            for (BrewingRecipe recipe : apotions)
+        if (outputId.equals("brewing") && getClass() == BrewingRecipeHandler.class) {// don't want subclasses getting a hold of this
+            for (NEIBrewingRecipe recipe : apotions)
                 arecipes.add(new CachedBrewingRecipe(recipe));
-        else
+            for (IBrewingRecipe recipe : BrewingRecipeRegistry.getRecipes())
+                if (recipe instanceof BrewingRecipe || recipe instanceof BrewingOreRecipe)
+                    arecipes.add(new CachedBrewingRecipe((AbstractBrewingRecipe<?>) recipe));
+        } else
             super.loadCraftingRecipes(outputId, results);
     }
 
     @Override
     public void loadCraftingRecipes(ItemStack result) {
-        if (result.getItem() != potionitem) return;
-        int damage = result.getItemDamage();
+        if (result.getItem() == Items.potionitem) {
+            int damage = result.getItemDamage();
+            for (NEIBrewingRecipe recipe : apotions)
+                if (recipe.output.item.getItemDamage() == damage)
+                    arecipes.add(new CachedBrewingRecipe(recipe));
+        }
 
-        for (BrewingRecipe recipe : apotions)
-            if (recipe.result.item.getItemDamage() == damage)
-                arecipes.add(new CachedBrewingRecipe(recipe));
+        for (IBrewingRecipe recipe : BrewingRecipeRegistry.getRecipes())
+            if (recipe instanceof BrewingRecipe || recipe instanceof BrewingOreRecipe)
+                if (NEIServerUtils.areStacksSameType(((AbstractBrewingRecipe<?>) recipe).output, result))
+                    arecipes.add(new CachedBrewingRecipe((AbstractBrewingRecipe<?>) recipe));
     }
 
     @Override
     public void loadUsageRecipes(ItemStack ingredient) {
-        if (ingredient.getItem() != potionitem && !ingredients.contains(ingredient)) return;
-
-        for (BrewingRecipe recipe : apotions)
-            if (NEIServerUtils.areStacksSameType(recipe.ingredient.item, ingredient) || NEIServerUtils.areStacksSameType(recipe.precursorPotion.item, ingredient))
-                arecipes.add(new CachedBrewingRecipe(recipe));
+        if (ingredient.getItem() == Items.potionitem || ingredients.contains(ingredient))
+            for (NEIBrewingRecipe recipe : apotions)
+                if (NEIServerUtils.areStacksSameType(recipe.ingredient.item, ingredient) || NEIServerUtils.areStacksSameType(recipe.input.item, ingredient))
+                    arecipes.add(new CachedBrewingRecipe(recipe));
+
+        label0: for (IBrewingRecipe recipe : BrewingRecipeRegistry.getRecipes())
+            if (recipe instanceof BrewingRecipe || recipe instanceof BrewingOreRecipe)
+                if (NEIServerUtils.areStacksSameType(((AbstractBrewingRecipe<?>) recipe).input, ingredient))
```

Comment:
The AbstractBrewingRecipe<?> cast is done 4 times, it's a bit verbose, make a local variable, perhaps named arecipe